### PR TITLE
Add instructions about OTA, Jobs, PKCS11 demos in README

### PR DESCRIPTION
**Pull request by aggarw13**
Update README with instructions for OTA and Jobs demos along with link to setup a **Thing resource** in their AWS account (required for AWS specific libraries)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -320,6 +327,57 @@ In order to set these configurations manually, edit `demo_config.h` in `demos/ht
 
 You can generate the presigned urls using [demos/http/common/src/presigned_urls_gen.py](demos/http/common/src/presigned_urls_gen.py). More info can be found [here](demos/http/common/src/README.md).
 
+#### Setup for AWS IoT Jobs demo
+
+1. The demo requires the Linux platform to contain curl and libmosquitto which can be installed with:
+
+```
+    apt install curl libmosquitto-dev
```
Comment (yourslab) [inline README.md:335]: It is not necessarily required. It will build it from source if not installed.

Comment (yourslab) [reply README.md:335]: Also not all Linux distros use apt

File: README.md [hunk 2]
```diff
@@ -320,6 +327,57 @@ In order to set these configurations manually, edit `demo_config.h` in `demos/ht
...
+		<li> Manage → Jobs → Create → Create a FreeRTOS OTA update job → Select `testclient` from the thing list</li>
+		<li> Sign a new firmware → Create a new profile → Select any SHA-ECDSA signing platform → Upload the code signing certificate(from prerequisites) and provide its path on the device.</li>
+		<li> Select the image → Select the bucket you created in prerequisites → Upload the binary build-dir-2/bin/ota_demo2</li>
+		<li> The path on device should be the complete path to place the executable and the binary name: eg /home/ubuntu/aws-iot-device-sdk-embedded-C-staging/build-dir/bin/ota_demo_core_mqtt2</li>
+		<li> Select the IAM role created in prerequisites</li>
+		<li> Create the Job
+		</ol>
+	</li>
+	<li>Run the initial executable again with the following command: `sudo ./ota_demo_core_mqtt or sudo ./ota_demo_core_http`</li>
+</ol>
```
Comment (yourslab) [inline README.md:379]: Why use HTML when you can use markdown here?

Comment (aggarw13) [reply README.md:379]: I have taken it from the doxygen page. Removing the HTML tags involved more work than just copying it

**Review by leegeth — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (aggarw13) [reply README.md:338]: Addressed

**Review by leegeth — DISMISSED** [review]

**Review by leegeth — APPROVED** [review]

**Review by n9wxu — APPROVED** [review]

**Pull request merged**